appserver/security/core-ee/src/main/java/com/sun/enterprise/security/ee/SecuritySniffer.java

```diff
@@ -80,19 +80,21 @@
     };
 
     public SecuritySniffer() {
-        super("security", "WEB-INF/web.xml", null);
-        
+        super("security", null, null);
     }
-    
-   /**
+
+    /**
      * Returns true if the passed file or directory is recognized by this
      * instance.
      *
      * @param location the file or directory to explore
      * @return true if this sniffer handles this application type
      */
     public boolean handles(ReadableArchive location) {
-        return (DeploymentUtils.isArchiveOfType(location, DOLUtils.warType(), habitat) || DeploymentUtils.isArchiveOfType(location, DOLUtils.earType(), habitat) || isJar(location));
+        ArchiveType archiveType = location.getExtraData(ArchiveType.class);
```

Comment:
There is another (inherited) handles method from GenericSniffer which takes a DeploymentContext argument, we can override that method in SecuritySniffer to add the additional condition for matching with the archive type before it invokes the handles(ReadableArchive). This way, we don't need to set the archive type on the ReadableArchive. 

   /**
     * Returns true if the passed file or directory is recognized by this
     * composite sniffer.
     * @param context deployment context
     * @return true if the location is recognized by this sniffer
     */
    public boolean handles(DeploymentContext context) {
        ArchiveType archiveType = habitat.getService(ArchiveType.class, context.getArchiveHandler().getArchiveType());
        if (archiveType != null && !supportsArchiveType(archiveType)) {
            return false;
        }
        return handles(context.getSource());
    }